llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -9636,9 +9636,27 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
                                        ? CmpInst::BAD_FCMP_PREDICATE
                                        : CmpInst::BAD_ICMP_PREDICATE;
 
-      return TTI->getCmpSelInstrCost(E->getOpcode(), OrigScalarTy,
-                                     Builder.getInt1Ty(), CurrentPred, CostKind,
-                                     VI);
+      InstructionCost ScalarCost = TTI->getCmpSelInstrCost(
+          E->getOpcode(), OrigScalarTy, Builder.getInt1Ty(), CurrentPred,
+          CostKind, VI);
+      auto IntrinsicAndUse = canConvertToMinOrMaxIntrinsic(VI);
```

Comment:
Is this any cleaner if we use a binding? (and below in GetVectorCost).
```
auto [MinMaxID, SelectOnly] = canConvertToMinOrMaxIntrinsic(VI);
```

Reply:
Done